### ISSUE TITLE
Why COVNet need BN for testing

### ISSUE THREAD

**Issue by kexianer** [bkong999/COVNet#17]
Why COVNet need BN for testing?
`    model.eval()
    for m in model.modules():
        if isinstance(m, nn.BatchNorm2d):
            m.train()
            m.weight.requires_grad = False
            m.bias.requires_grad = False`

I have tried to get rid of BN in the validation， but I got bad result. I am puzzled about the reason for this result.

**Comment by bkong999** [comment]
> Why COVNet need BN for testing?
> ` model.eval() for m in model.modules(): if isinstance(m, nn.BatchNorm2d): m.train() m.weight.requires_grad = False m.bias.requires_grad = False`
> 
> I have tried to get rid of BN in the validation， but I got bad result. I am puzzled about the reason for this result.

I don't know what you mean by why you need BN for testing. You need to have the same network during the training and testing stage. The only difference is that the BN will behave differently. 

I guess you mean "why you set the BN layers like this?" Well, basically, we are freezing the BN layer as our batch size is only one. If we train the BN layer with very small batch size, the performance will be severely degraded. Please refer to this link for a more detailed explanation: https://discuss.pytorch.org/t/how-to-train-with-frozen-batchnorm/12106/8.
